### ISSUE TITLE
A LoRA run under FSDP writes a 37 GB consolidated base model into every checkpoint (adapter is 131 MB)

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#352]
Measured at 70B on 8x H100 while running #41. Full record: `benchmarks/gate-h100-validation.md`, STEP 28.

A `use_fsdp2_compile` + LoRA r16 run on Qwen2.5-72B wrote this into a single checkpoint directory:

```
adapter_model.safetensors        131 118 496   (131 MB — the thing being trained)
optimizer.bin                    262 476 043
pytorch_model_fsdp.bin        36 981 387 757   (37 GB — a full consolidated copy of the frozen base)
rng_state_0..7.pth
```

The base is frozen; 0.18% of the model is being trained. Writing a full copy of it per checkpoint is 280x the useful artifact.

## Why it matters

On a box with 250 GB free that is seven checkpoints from a full disk, and a full disk stops the run rather than degrading it. It also makes `save_steps` effectively unusable on any large model under FSDP: a user who sets a sensible cadence fills the volume.

## Cause

Accelerate's FSDP state-dict handling writes a `FULL_STATE_DICT` for the wrapped model alongside the PEFT adapter. HF's `save_only_model=True` skips optimizer/scheduler/RNG but not this; the relevant knob is the FSDP state-dict type, or not saving the base at all for a PEFT model.

## Suggested fix

When the model is a `PeftModel` and the base is frozen, do not write the base into checkpoints — the adapter plus config is the complete artifact. Failing that, default `save_only_model=True` on the FSDP path and document it.

## Acceptance criteria

- [ ] A LoRA + FSDP checkpoint directory contains no full-model file
- [ ] Test asserts checkpoint size is O(adapter), not O(base), for a PEFT model under an FSDP preset
- [ ] Resume still works from such a checkpoint

**Comment by MakazhanAlpamys** [comment]
Reopening deliberately. PR #591 (`14a93ec`) auto-closed this via its `Fixes #352` keyword, but I told @AmirF194 on the PR that I would keep it open for the one acceptance criterion that is not met, so I am honouring that rather than quietly letting it close.

**What #591 did close.** The `accelerate` floor is now `>=0.27.0`, the first release whose `save_fsdp_model` / `load_fsdp_model` accept `adapter_only`. Verified against the real release tags: v0.26.0's signature is `save_fsdp_model(fsdp_plugin, accelerator, model, output_dir, model_index=0)`; v0.27.0 adds the parameter there and on `load_fsdp_model`, `load_fsdp_optimizer` and `_get_model_state_dict`. Under the old `>=0.25.0` floor pip could legally resolve 0.25 or 0.26, where transformers' feature detection returns `{}` and the full base is written again.

**Correcting the record on what the 37 GB measurement was.** The STEP 28 run in `benchmarks/gate-h100-validation.md` had **accelerate 1.14.0**, which already carried `adapter_only`, and **transformers 4.57.6**, which has no `get_fsdp_ckpt_kwargs` at all. So the measured regression was on the transformers side and is already closed by the `transformers>=5.16.1` floor on main; #591 closed the remaining resolvable way to defeat that. Worth stating plainly so nobody later reads this issue as "the accelerate bump fixed the 37 GB write".

**What is still open — criterion 3 only.**

- [x] A LoRA + FSDP checkpoint directory contains no full-model file — substantively. `pytorch_model_fsdp.bin` is still written; it now holds adapter tensors only.
- [x] Test asserts checkpoint size is O(adapter), not O(base) — as a unit test on accelerate's own state-dict helper (16,384 params against 4,210,688), not on a real checkpoint directory.
- [ ] **Resume still works from such a checkpoint.** Untested.

On that last one, @AmirF194 made an argument I accept and want on the record rather than buried in a PR thread: Soup imports `transformers.Trainer` unmodified and the save/load call happens inside accelerate, so a `Trainer.train()` round trip would mostly exercise upstream internals, and making it real needs an actual FSDP process group, which is not CPU-safe. I told them not to build it. So this stays open as a **known gap, not a request** — if someone with multi-GPU CI wants it, the shape is a single-process gloo FSDP group, and it needs GPU CI or a longer-running job rather than the existing CPU suite.

Separately, a guard gap I found while merging and am recording here rather than filing noise: `tests/test_issue502_qwen35_transformers5.py::test_doctor_training_bounds_match_declared_extra` pins `commands/doctor.py`'s `DEPS` floors against the declared extra, but only for the three packages in `_RUNTIME_FLOORS` (transformers, trl, peft). `accelerate` is not among them, which is why reverting the `doctor.py` row to `0.25.0` still passes the suite. Adding it to `_RUNTIME_FLOORS` would also change the `_REQUIRES_TRAINING_FLOOR` skip behaviour, so it wants a separate assertion rather than a one-line addition. Small, self-contained, and a reasonable first issue for someone.
